addons/hw_scale/controllers/main.py

```diff
@@ -222,32 +265,34 @@ def get_device(self):
                         continue
                     path = self.input_dir + device
                     for protocol in SCALE_PROTOCOLS:
-                        _logger.info('Probing %s with protocol %s', path, protocol)
-                        connection = serial.Serial(path,
-                                                   baudrate=protocol.baudrate,
-                                                   bytesize=protocol.bytesize,
-                                                   stopbits=protocol.stopbits,
-                                                   parity=protocol.parity,
-                                                   timeout=1,      # longer timeouts for probing
-                                                   writeTimeout=1) # longer timeouts for probing
-                        connection.write(protocol.weightCommand + protocol.commandTerminator)
-                        time.sleep(protocol.commandDelay)
-                        answer = self._get_raw_response(connection)
-                        weight, weight_info, status = self._parse_weight_answer(protocol, answer)
-                        if status:
-                            _logger.info('Probing %s: no valid answer to protocol %s', path, protocol.name)
-                        else:
-                            _logger.info('Probing %s: answer looks ok for protocol %s', path, protocol.name)
-                            self.path_to_scale = path
-                            self.protocol = protocol
-                            self.set_status(
-                                'connected',
-                                'Connected to %s with %s protocol' % (device, protocol.name)
-                            )
-                            connection.timeout = protocol.timeout
-                            connection.writeTimeout = protocol.writeTimeout
-                            hw_proxy.rs232_devices[path] = DRIVER_NAME
-                            return connection
+                        for tryCounter in range(protocol.maxConnectionTry):
```

Comment:
Hello,

You have added three new required parameters to ('outputTerminator', 'commandSupport', 'maxConnectionTry') could you please remove them from the existing protocols, and handle them in the code with default values to avoid breaking the custom protocols developed by partners.

Could you please also tell me why do you have to try to connect three times to your scale?

Thanks

Reply:
Hello,
These scales with continuous weight output gives us some random character for first 1-2 times, afterwards, it gives us the perfect weight output. So, we make it try to get the output for 3 times.

Reply:
OK, no problem, just apply the first part of the comment then.

Reply:
I've checked with ODO, and indeed you have to add keys in the namedtuple, but you can remove the key 'commandSupport' by changing the conditions in your 'if' statements.